test/integration/apis/issue.test.ts

```diff
@@ -87,44 +79,101 @@ describe("issue", () => {
             issueAPI.setAccount(alice);
             ImportMock.mockFunction(issueAPI, "isExecutionSucessful", true);
-            const isExecutionCorrect = await issueAPI.execute(issueId, txId, txBlockHeight, merkleProof, rawTx);
+            const isExecutionCorrect = await issueAPI.execute(issueId, txId, merkleProof, rawTx);
             assert.isTrue(isExecutionCorrect);
         });
 
         it("should consider execution failed if `isExecutionSucessful` returns false", async () => {
-            const { issueId, txId, txBlockHeight, merkleProof, rawTx } = makeExecutionData();
+            const { issueId, txId, merkleProof, rawTx } = makeExecutionData();
             issueAPI.setAccount(alice);
             ImportMock.mockFunction(issueAPI, "isExecutionSucessful", false);
-            const isExecutionCorrect = await issueAPI.execute(issueId, txId, txBlockHeight, merkleProof, rawTx);
+            const isExecutionCorrect = await issueAPI.execute(issueId, txId, merkleProof, rawTx);
             assert.isFalse(isExecutionCorrect);
         });
+
+        it("should request and execute issue", async () => {
+            const btcCore = new DefaultBTCCoreAPI("http://0.0.0.0:3002");
+            btcCore.initializeClientConnection("regtest", "0.0.0.0", "rpcuser", "rpcpassword", "18443", "Alice");
+            let initialBalance = await treasuryAPI.balancePolkaBTC(api.createType("AccountId", alice.address));
+            const blocksToMine = 3;
+            keyring = new Keyring({ type: "sr25519" });
+            alice = keyring.addFromUri("//Alice");
+            dave = keyring.addFromUri("//Dave");
+
+            // request issue
+            issueAPI.setAccount(alice);
+            const amountAsBtcString = "0.001";
+            const amountAsSatoshiString = btcToSat(amountAsBtcString);
+            if (amountAsSatoshiString === undefined) {
+                fail();
+            }
+            const amountAsSatoshi = api.createType("Balance", amountAsSatoshiString);
+            const requestResult = await issueAPI.request(amountAsSatoshi);
+            assert.isTrue(requestResult.hash.length > 0);
+
+            // send btc tx
+            const data = stripHexPrefix(requestResult.hash.toString());
+            const vaultBtcAddress = encodeBtcAddress(requestResult.vault.wallet.address, bitcoin.networks.regtest);
+            if (vaultBtcAddress === undefined) {
+                throw new Error("Undefined vault address returned from RequestIssue");
+            }
+            const txData = await btcCore.broadcastOpReturnTx(vaultBtcAddress, amountAsBtcString, data);
+            await btcCore.mineBlocks(blocksToMine);
+
+            // check issuing worked
+            let finalBalance = await treasuryAPI.balancePolkaBTC(api.createType("AccountId", alice.address));
+            assert.isTrue(finalBalance.toBn().sub(initialBalance.toBn()).eq(new BN(amountAsSatoshiString)));
+        });
```

Comment:
We need to test the execute isue function here as well. For this set the `--no-issue-execution` flag on the vault client in the docker container.